common/lib/transport/connectionmanager.js

```diff
@@ -809,70 +785,217 @@ var ConnectionManager = (function() {
 		this.enactStateChange(change);
 	};
 
-	ConnectionManager.prototype.connectImpl = function() {
-		var state = this.state;
-		if(state == this.states.closing || state == this.states.closed || state == this.states.failed) {
-			Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.connectImpl()', 'abandoning connection attempt; state = ' + state.state);
-			return;
-		}
 
-		Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.connectImpl()', 'starting connection');
+	ConnectionManager.prototype.startConnect = function() {
+		var auth = this.realtime.auth,
+			self = this;
+
+		var connect = function() {
+			self.getTransportParams(function(transportParams) {
+				self.connectImpl(transportParams);
+			});
+		};
+
+		Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.startConnect()', 'starting connection');
 		this.startSuspendTimer();
 		this.startTransitionTimer(this.states.connecting);
 
-		var self = this;
-		var auth = this.realtime.auth;
-		var connectErr = function(err) {
-			err = ErrorInfo.fromValues(err);
-			Logger.logAction(Logger.LOG_ERROR, 'ConnectionManager.connectImpl()', 'Connection attempt failed with error; err = ' + err.toString());
-			var state = self.state;
-			if(state == self.states.closing || state == self.states.closed || state == self.states.failed) {
-				/* do nothing */
-				return;
+		if(auth.method == 'basic') {
```

Comment:
Oops, yep

(it's the switching back and forth from js to elixir (which uses == and double-quotes for utf8 strings) that does it :) )